### PR TITLE
Document strings api #179

### LINKED ISSUE #179: Document strings API
The current strings API (see `zio.redis.api.Strings`) exposes functions with argument names that are generic and meaningless. I propose to rename those to match those used in the [official documentation](https://redis.io/commands). Additionally, it would be great to provide a small, single-sentence scaladoc for them.

### PR DESCRIPTION
**Pull request by justinhj**
Resolves #179

### CODE REVIEW

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 1]
```diff
@@ -10,68 +10,116 @@ import zio.{ Chunk, ZIO }
...
+    MSet.run((firstKeyValue, restKeyValues.toList))
+
+  // Set multiple keys to multiple values only if none of the keys exist
+  final def mSetNx(
+    firstKeyValue: (String, String),
+    restKeyValues: (String, String)*
+  ): ZIO[RedisExecutor, RedisError, Boolean] =
+    MSetNx.run((firstKeyValue, restKeyValues.toList))
+
+  // Set the value and expiration in milliseconds of a key
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:87]: In order to have a valid scaladoc comment, you should use `/** <comment here> */` format as you've already used below. Also, try using `@param` to describe method parameters and `@return` to describe a return value. I suggest that you don't use `@return` when the returned ZIO effect has a value of type `Unit`.

Here is one example:
```suggestion
 /** Sets a value and expiration in milliseconds of the specified key.
  *
  *  @param key Redis key identifier
  *  @param milliseconds expire time that is expressed in milliseconds
  *  @param value key value to be set
  */
```

For more details you can read the [documentation](https://docs.scala-lang.org/style/scaladoc.html).

Comment (justinhj) [reply redis/src/main/scala/zio/redis/api/Strings.scala:87]: Sure I will update it shortly.

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 2]
```diff
@@ -10,68 +10,235 @@ import zio.{ Chunk, ZIO }
...
-    MSet.run((a, as.toList))
-
-  final def mSetNx(a: (String, String), as: (String, String)*): ZIO[RedisExecutor, RedisError, Boolean] =
-    MSetNx.run((a, as.toList))
-
-  final def pSetEx(a: String, b: Duration, c: String): ZIO[RedisExecutor, RedisError, Unit] = PSetEx.run((a, b, c))
-
+    key: String,
+    firstCommand: BitFieldCommand,
+    restCommands: BitFieldCommand*
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:41]: I'd rename to `command` and `commands`.

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 3]
```diff
@@ -10,68 +10,235 @@ import zio.{ Chunk, ZIO }
...
+   * @param destKey Key of destination string to store the result
+   * @param firstSrcKey First source key to apply the operation to
+   * @param restSrcKeys Subsequent source keys to apply the operation to
+   * @return Returns size of the string stored in the destination key, that is equal to the size of the longest input string
+   */
+  final def bitOp(
+    operation: BitOperation,
+    destKey: String,
+    firstSrcKey: String,
+    restSrcKeys: String*
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:56]: I'd rename to `srcKey` and `srcKeys`.

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 4]
```diff
@@ -10,68 +10,235 @@ import zio.{ Chunk, ZIO }
...
+  final def incrByFloat(key: String, increment: Double): ZIO[RedisExecutor, RedisError, String] =
+    IncrByFloat.run((key, increment))
+
+  /** Get all the values of the given keys
+   *
+    * @param firstKey First key to get
+   * @param restKeys Subsequent keys to get
+   * @return Returns the values of the given keys
+   */
+  final def mGet(firstKey: String, restKeys: String*): ZIO[RedisExecutor, RedisError, Chunk[Option[String]]] =
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:148]: I'd rename to `key` and `keys`.

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 5]
```diff
@@ -10,68 +10,235 @@ import zio.{ Chunk, ZIO }
...
+    MGet.run((firstKey, restKeys.toList))
+
+  /** Set multiple keys to multiple values
+   *
+    * @param firstKeyValue Tuple of key and value, first one to set
+   * @param restKeyValues Subsequent tuples of key values
+   */
+  final def mSet(
+    firstKeyValue: (String, String),
+    restKeyValues: (String, String)*
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:158]: I'd rename to `keyValue` and `keyValues`.

File: redis/src/main/scala/zio/redis/api/Strings.scala [hunk 6]
```diff
@@ -10,68 +10,235 @@ import zio.{ Chunk, ZIO }
...
+
+  /** Set multiple keys to multiple values only if none of the keys exist
+   *
+    * @param firstKeyValue First key value to set
+   * @param restKeyValues Subsequent key values to set
+   * @return 1 if the all the keys were set. 0 if no key was set (at least one key already existed)
+   */
+  final def mSetNx(
+    firstKeyValue: (String, String),
+    restKeyValues: (String, String)*
```
Comment (anovakovic01) [inline redis/src/main/scala/zio/redis/api/Strings.scala:170]: I'd rename to `keyValue` and `keyValues`.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]
Good job @justinhj, almost done.

File: redis/src/main/scala/zio/redis/api/Strings.scala (hunk 2 above)
Comment (aleksandarskrbic) [reply redis/src/main/scala/zio/redis/api/Strings.scala:41]: @anovakovic01 I like your naming suggestion more, but this style already passed some earlier [PR](https://github.com/zio/zio-redis/pull/185/files). @mijicd We should choose one style and stick to it?

Comment (anovakovic01) [reply redis/src/main/scala/zio/redis/api/Strings.scala:41]: Good catch. I agree that we should stick to one style only. @mijicd should decide which one.

Comment (aleksandarskrbic) [reply redis/src/main/scala/zio/redis/api/Strings.scala:41]: I've followed the style of that merged PR, but I could change my PR.

Comment (mijicd) [reply redis/src/main/scala/zio/redis/api/Strings.scala:41]: Let's go with the proposal from @anovakovic01

Comment (justinhj) [reply redis/src/main/scala/zio/redis/api/Strings.scala:41]: I will update the names to remove the first and rest prefixes. I will need to open a new PR to do the same in the Sets API

**Review by anovakovic01 — APPROVED** [review]

**Comment by anovakovic01** [comment]
@justinhj Nice work. We agreed to modify Scaladoc comment format so that every comment has the first line empty.

**Pull request merged**